src/utils/survey-event-receiver.ts

```diff
@@ -0,0 +1,60 @@
+import { Survey } from '../posthog-surveys-types'
+
+export class SurveyEventReceiver {
+    private eventRegistry: Map<string, string[]>
+    private surveysActivatedKey = 'surveysActivated'
```

Comment:
Worth adding posthog to the key here to scope it better, if we end up using the window.sessionStorage (see below)